### PR TITLE
feat(frontend): migrate to vite :)

### PR DESCRIPTION
**Pull request by zaxutic**
yeet out CRA/react-scripts and switch to [vite](https://vitejs.dev/) for *blazingly fast* dev server startup and cleaner configuration, and to avoid any future potential problems because craco cringe (and react-scripts often lags behind with supporting newer libraries - e.g., tailwind previously requiring CRA users to install a compatibility build for postcss 7)

since vite by default reasonably [does not let you use jsx in `.js` files](https://twitter.com/youyuxi/status/1362050255009816577), we change all of our files that use this to the appropriate extension. since this will introduce merge conflicts with everything currently open for frontend changes, this is a draft PR until we get all those merged

this will likely require @lhjt to modify the environment variables for staging to use `VITE_` instead of `REACT_APP_` as well

### CODE REVIEW

**Comment by imagine-hussain** [comment]
lgtm

**Comment by zaxutic** [comment]
i have been informed (and tested) that git is in fact smart enough to properly handle merges involving renames (merging a branch that renames a file, then merging a branch that modifies the old file) so we actually do not have to wait until other PRs get merged before this one

File: frontend/package.json
Comment (zaxutic) [inline frontend/package.json:35]: we don't have any tests anyway lol but if we start doing tests we can use [vitest](https://vitest.dev/)

File: frontend/package.json
Comment (lhjt) [inline frontend/package.json:30]: Are these required during runtime or can they be moved to devDependencies?

Comment (zaxutic) [reply frontend/package.json:30]: good point, these don't need to be here

**Review by lhjt — APPROVED** [review]
Let me know before you'd like to merge this in so that @csesoc/technical can update the pipeline variables; otherwise this looks good to me

**Review by lhjt — APPROVED** [review]

**Pull request merged**
